File: android/app/src/main/java/com/docavailable/ScreenshotPreventionModule.java
package com.docavailable;

import android.app.Activity;
import android.view.WindowManager;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;
import com.facebook.react.bridge.Promise;

public class ScreenshotPreventionModule extends ReactContextBaseJavaModule {
    
    public ScreenshotPreventionModule(ReactApplicationContext reactContext) {
        super(reactContext);
    }

    @Override
    public String getName() {
        return "ScreenshotPreventionModule";
    }

    @ReactMethod
    public void setSecureFlag(boolean enabled, Promise promise) {
        try {
            Activity currentActivity = getCurrentActivity();
            if (currentActivity != null) {
                if (enabled) {
                    // Enable FLAG_SECURE to prevent screenshots - will show black screen
                    currentActivity.getWindow().setFlags(
                        WindowManager.LayoutParams.FLAG_SECURE,
                        WindowManager.LayoutParams.FLAG_SECURE
                    );
                    // Also prevent screen recording and other capture methods
                    currentActivity.getWindow().addFlags(WindowManager.LayoutParams.FLAG_SECURE);
                    // Additional security flags
                    currentActivity.getWindow().addFlags(WindowManager.LayoutParams.FLAG_HARDWARE_ACCELERATED);
                    System.out.println("🔒 [ScreenshotPrevention] Android FLAG_SECURE enabled - screenshots will show black screen");
                } else {
                    // Disable FLAG_SECURE to allow screenshots
                    currentActivity.getWindow().clearFlags(
                        WindowManager.LayoutParams.FLAG_SECURE
                    );
                    System.out.println("🔓 [ScreenshotPrevention] Android FLAG_SECURE disabled - screenshots allowed");
                }
                promise.resolve(true);
            } else {
                System.out.println("❌ [ScreenshotPrevention] No current activity found");
                promise.reject("NO_ACTIVITY", "No current activity found");
            }
        } catch (Exception e) {
            System.out.println("❌ [ScreenshotPrevention] Failed to set secure flag: " + e.getMessage());
            promise.reject("ERROR", "Failed to set secure flag: " + e.getMessage());
        }
    }

    @ReactMethod
    public void isSecureFlagEnabled(Promise promise) {
        try {
            Activity currentActivity = getCurrentActivity();
            if (currentActivity != null) {
                boolean isEnabled = (currentActivity.getWindow().getAttributes().flags 
                    & WindowManager.LayoutParams.FLAG_SECURE) != 0;
                promise.resolve(isEnabled);
            } else {
                promise.reject("NO_ACTIVITY", "No current activity found");
            }
        } catch (Exception e) {
            promise.reject("ERROR", "Failed to check secure flag: " + e.getMessage());
        }
    }
}

File: android/app/src/main/java/com/docavailable/app/ScreenshotPreventionModule.kt
package com.docavailable.app

import android.app.Activity
import android.view.WindowManager
import com.facebook.react.bridge.Promise
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod
import com.facebook.react.bridge.UiThreadUtil

class ScreenshotPreventionModule(reactContext: ReactApplicationContext) :
    ReactContextBaseJavaModule(reactContext) {

    override fun getName(): String {
        return "ScreenshotPreventionModule"
    }

    @ReactMethod
    fun setSecureFlag(enabled: Boolean, promise: Promise) {
        try {
            UiThreadUtil.runOnUiThread {
                try {
                    val activity: Activity? = currentActivity
                    if (activity == null) {
                        promise.reject("ERROR", "Activity is null - cannot set secure flag")
                        return@runOnUiThread
                    }

                    val window = activity.window
                    if (window == null) {
                        promise.reject("ERROR", "Window is null - cannot set secure flag")
                        return@runOnUiThread
                    }

                    if (enabled) {
                        // Enable FLAG_SECURE to prevent screenshots and screen recording
                        window.setFlags(
                            WindowManager.LayoutParams.FLAG_SECURE,
                            WindowManager.LayoutParams.FLAG_SECURE
                        )
                        android.util.Log.d("ScreenshotPrevention", "✅ FLAG_SECURE enabled - screenshots will show black screen")
                        promise.resolve("Screenshot prevention enabled")
                    } else {
                        // Disable FLAG_SECURE to allow screenshots
                        window.clearFlags(WindowManager.LayoutParams.FLAG_SECURE)
                        android.util.Log.d("ScreenshotPrevention", "🔓 FLAG_SECURE disabled - screenshots allowed")
                        promise.resolve("Screenshot prevention disabled")
                    }
                } catch (e: Exception) {
                    android.util.Log.e("ScreenshotPrevention", "❌ Error setting secure flag: ${e.message}", e)
                    promise.reject("ERROR", "Failed to set secure flag: ${e.message}", e)
                }
            }
        } catch (e: Exception) {
            android.util.Log.e("ScreenshotPrevention", "❌ Error in setSecureFlag: ${e.message}", e)
            promise.reject("ERROR", "Failed to execute setSecureFlag: ${e.message}", e)
        }
    }

    @ReactMethod
    fun isSecureFlagEnabled(promise: Promise) {
        try {
            val activity: Activity? = currentActivity
            if (activity == null) {
                promise.reject("ERROR", "Activity is null")
                return
            }

            val window = activity.window
            if (window == null) {
                promise.reject("ERROR", "Window is null")
                return
            }

            val flags = window.attributes.flags
            val isSecure = (flags and WindowManager.LayoutParams.FLAG_SECURE) != 0
            promise.resolve(isSecure)
        } catch (e: Exception) {
            promise.reject("ERROR", "Failed to check secure flag: ${e.message}", e)
        }
    }
}

